packages/core/i18n/core-i18n-server-internal/src/routes/translations.ts

```diff
@@ -6,54 +6,81 @@
  * Side Public License, v 1.
  */
 
-import { createHash } from 'crypto';
 import { i18n } from '@kbn/i18n';
 import { schema } from '@kbn/config-schema';
 import type { IRouter } from '@kbn/core-http-server';
 
+const MINUTE = 60;
+const HOUR = 60 * MINUTE;
+const DAY = 24 * HOUR;
+
 interface TranslationCache {
   translations: string;
   hash: string;
 }
 
-export const registerTranslationsRoute = (router: IRouter, locale: string) => {
+export const registerTranslationsRoute = ({
+  router,
+  locale,
+  translationHash,
+  isDist,
+}: {
+  router: IRouter;
+  locale: string;
+  translationHash: string;
+  isDist: boolean;
+}) => {
   let translationCache: TranslationCache;
 
-  router.get(
-    {
-      path: '/translations/{locale}.json',
-      validate: {
-        params: schema.object({
-          locale: schema.string(),
-        }),
-      },
-      options: {
-        access: 'public',
-        authRequired: false,
-      },
-    },
-    (ctx, req, res) => {
-      if (req.params.locale.toLowerCase() !== locale.toLowerCase()) {
-        return res.notFound({
-          body: `Unknown locale: ${req.params.locale}`,
-        });
-      }
-      if (!translationCache) {
-        const translations = JSON.stringify(i18n.getTranslation());
-        const hash = createHash('sha1').update(translations).digest('hex');
-        translationCache = {
-          translations,
-          hash,
-        };
-      }
-      return res.ok({
-        headers: {
-          'content-type': 'application/json',
-          'cache-control': 'must-revalidate',
-          etag: translationCache.hash,
+  ['/translations/{locale}.json', `/translations/${translationHash}/{locale}.json`].forEach(
```

Comment:
I kept the old route for two reasons:
- functional tests against the endpoint (it's a pain to retrieve the translation hash from the API integration tests...)
- BWC: some integrations/customers *might* be reaching this endpoint, you never know

Reply:
This path is not a public API so i dont think we should worry about custom integrations especially since i dont see any use case for this.. Maybe worth creating an issue to remove it later on in a separate PR